aten/src/ATen/native/sparse/cuda/SparseBlas.cpp

```diff
@@ -83,16 +83,6 @@ Tensor& addmm_out_sparse_compressed_cuda(
     const Scalar& beta,
     const Scalar& alpha,
     Tensor& result) {
-
```

Comment:
The checks here incorrectly block some layout combinations which are possible.

In terms of separating responsibilities this layer does size/device compatibility checks only. All layout combination checking and conversions are done at the next layer.